src/OrchardCore.Modules/OrchardCore.Users/Workflows/Activities/ValidateUserTask.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Security.Services;
+using OrchardCore.Workflows.Abstractions.Models;
+using OrchardCore.Workflows.Activities;
+using OrchardCore.Workflows.Models;
+
+namespace OrchardCore.Users.Workflows.Activities
+{
+    public class ValidateUserTask : TaskActivity
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly UserManager<IUser> _userManager;
+        private readonly IStringLocalizer S;
+        private readonly IRoleService _roleService;
+
+        public ValidateUserTask(UserManager<IUser> userManager, IHttpContextAccessor httpContextAccessor, IStringLocalizer<ValidateUserTask> localizer, IRoleService roleService)
+        {
+            _httpContextAccessor = httpContextAccessor;
+            _userManager = userManager;
+            _roleService = roleService;
+            S = localizer;
+        }
+
+        public override string Name => nameof(ValidateUserTask);
+
+        public override LocalizedString Category => S["User"];
+
+        public bool SetUser
+        {
+            get => GetProperty(() => true);
+            set => SetProperty(value);
+        }
+
+        public IEnumerable<string> Roles
+        {
+            get => GetProperty(() => new List<string>());
+            set => SetProperty(value);
+        }
+
+        public override LocalizedString DisplayText => S["Validate User Task"];
+
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return Outcomes(S["Anonymous"], S["Authenticated"], S["InRole"]);
+        }
+
+        public async override Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            var user = _httpContextAccessor.HttpContext.User;
+            var isAuthenticated = user?.Identity?.IsAuthenticated;
+
+            if (isAuthenticated == true)
+            {
+                if (SetUser)
+                {
+                    workflowContext.Properties["User"] = user;
```

Comment:
I'm not against but do we really need the `SetUser` property?

There is a WF `SetPropertyTask` that uses a javascript expression, and there is an `httpContext()` script `GlobalMethod`, so a `SetPropertyTask` with the following expression may work `httpContext().User` (needs to be tried)

Note: We have an `is_in_role` liquid filter, having the same related `GlobalMethod` for the scripting engine would allow us to check if an user is in roles by just using a WF `IfElseTask`